### DISCUSSION TITLE
Replace knob with interactive button by addon-controls

### DISCUSSION THREAD

**Discussion by dpeger** [storybookjs/storybook#15779] [Help] [answered]
I'm currently using knobs in my project and (as knobs are deprecated) want to switch to controls. This works for the standard cases where knobs are used to provide inputs for components. I however have a story where I used knobs to push items to an array that is used as component input:

```
export const defaultElements: string[] = ["test1", "test2", "test3"];

const addElementAction = () => {
  defaultElements.push("test");
};

export const Primary = () => ({
  component: ListComponent,
  props: {
    elements: defaultElements,
    addMessage: button('Add Element', addElementAction),
  },
});
```

Is it possible to replace this button knob with controls? Or is there another addon that offers such a functionality?

**Comment by flo-sch** [comment]
I think this may be a use-case for #11984

**Reply by shilman** [reply]
Please upvote https://github.com/storybookjs/storybook/issues/11971

**Reply by dpeger** [reply]
I added a 👍 and my use case to issue https://github.com/storybookjs/storybook/issues/11971

thanks

**Comment by dpeger** [comment, marked as answer]
Doesn't seem to be possible as of now. But corresponding use case is covered by issue #11971.
